app/services/courses_query_logger.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class CoursesQueryLogger
+  def initialize(applied_filters, scope, explain: false)
+    @applied_filters = applied_filters
+    @scope = scope
+    @log_tag = self.class.name
+    @logger = Rails.logger
+    @explain = explain
+  end
+
+  def call
+    log_filters
+    log_query
+  end
+
+  private
+
+  def log_filters
+    return unless Rails.env.local?
```

Comment:
This and line 35 could be squashed into a single call in `#call` itself.

Reply:
Done